### ISSUE TITLE
undefined-variable or used-before-assignment should be reported when variable is declared with a type annotation but not assigned to before use

### ISSUE THREAD

**Issue by waleedmebane** [PyCQA/pylint#5140]
### Bug description

```python
x: int
print(x)
```

Pylint does not report an error for the code above.

This code:
```
# x: int
print(x)
```
produces:

uninitialized_variable_test.py:2:6: E0602: Undefined variable 'x' (undefined-variable)

This code:
```
x
print(x)
```
produces:

uninitialized_variable_test.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
uninitialized_variable_test.py:1:0: E0602: Undefined variable 'x' (undefined-variable)
uninitialized_variable_test.py:2:6: E0602: Undefined variable 'x' (undefined-variable)

This code:
```
x: int
print(x)
x = 0
```

produces no error.

This code:
```
# x: int
print(x)
x = 0
```
produces:
uninitialized_variable_test.py:2:6: E0601: Using variable 'x' before assignment (used-before-assignment)

Python produces a runtime error for all three cases:
NameError: name 'x' is not defined

(FYI, Mypy and pytype also do not catch this error.)
```


### Configuration

_No response_

### Command used

```shell
pylint uninitialized_variable_test.py
```

### Pylint output

```shell
----------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: -45.00/10, +55.00)
```

### Expected behavior

Should get an undefined variable error in the same cases in which one would get the error without the variable having been declared but not assigned, OR it should produce a used-before-assignment error.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.2
Python 3.9.5 (default, Jul 19 2021, 01:21:38) 
[GCC 5.4.0 20160609]
```

### OS / Environment

Ubuntu 16.04
